core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -0,0 +1,220 @@
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging requests.
+     */
+    public LogLevel getRequestLogLevel() {
+        return requestLogLevel;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     * If unset, will use {@link LogLevel#TRACE}.
+     */
+    public T successfulResponseLogLevel(LogLevel successfulResponseLogLevel) {
+        this.successfulResponseLogLevel =
+                requireNonNull(successfulResponseLogLevel, "successfulResponseLogLevel");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     */
+    public LogLevel getSuccessfulResponseLogLevel() {
+        return successfulResponseLogLevel;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     * If unset, will use {@link LogLevel#WARN}.
+     */
+    public T failureResponseLogLevel(LogLevel failedResponseLogLevel) {
+        this.failedResponseLogLevel = requireNonNull(failedResponseLogLevel, "failedResponseLogLevel");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     */
+    public LogLevel getFailedResponseLogLevel() {
+        return failedResponseLogLevel;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize request headers before logging. It is common to have the
+     * {@link Function} remove sensitive headers, like {@code Cookie}, before logging. If unset, will use
+     * {@link Function#identity()}.
+     */
+    public T requestHeadersSanitizer(Function<HttpHeaders, HttpHeaders> requestHeadersSanitizer) {
+        this.requestHeadersSanitizer = requireNonNull(requestHeadersSanitizer, "requestHeadersSanitizer");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link Function} to use to sanitize request headers before logging.
+     */
+    public Function<HttpHeaders, HttpHeaders> getRequestHeadersSanitizer() {
+        return requestHeadersSanitizer;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize request content before logging. It is common to have the
+     * {@link Function} remove sensitive content, such as an GPS location query, before logging. If unset,
+     * will use {@link Function#identity()}.
+     */
+    public T requestContentSanitizer(Function<Object, Object> requestContentSanitizer) {
+        this.requestContentSanitizer = requireNonNull(requestContentSanitizer, "requestContentSanitizer");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link Function} to use to sanitize request content before logging.
+     */
+    public Function<Object, Object> getRequestContentSanitizer() {
```

Comment:
- Remove `get`?
- protected? (a subclass can override it to make it accessible from the classes from the same package)